lib/shopify_cli/environment.rb

```diff
@@ -106,7 +106,13 @@ def self.infer_spin?(env_variables: ENV)
 
     def self.spin_url(env_variables: ENV)
       if infer_spin?(env_variables: env_variables)
-        %x(spin info fqdn 2> /dev/null).strip
+        instances = JSON.parse(%x(exe/spin list --json --output created,fqdn))
```

Comment:
>Having another environment variable to set isn't ideal but I like the flexibility there instead of just picking the one I created last.

You should still have that flexibility- this is just a default. When I go to fix the merge conflict I'll keep in the `SPIN_INSTANCE` logic, and just have this as the backup case.